C/c4Log.cc

```diff
@@ -220,30 +220,50 @@ void c4log_initConsole(C4LogLevel level) noexcept {
 
 #pragma mark - FILE LOGGING:
 
-static C4LogObserver* sDefaultLogFiles;
-static C4LogLevel     sDefaultLogFilesLevel = kC4LogNone;
+static C4LogObserver*                                           sDefaultLogFiles;
+static C4LogLevel                                               sDefaultLogFilesLevel = kC4LogNone;
+static std::unique_ptr<C4LogObserver, void (*)(C4LogObserver*)> stagedDefaultLogFiles{
+        nullptr, [](C4LogObserver* p) { c4logobserver_release(p); }};
 
 static void endFileLogging() {
     if ( sDefaultLogFiles ) {
         c4log_removeObserver(sDefaultLogFiles);
         c4logobserver_release(sDefaultLogFiles);
         sDefaultLogFiles = nullptr;
     }
+    // Also clear the stagedDefaultLogFiles
+    stagedDefaultLogFiles = nullptr;
 }
 
 bool c4log_writeToBinaryFile(C4LogFileOptions options, C4Error* outError) noexcept {
-    if ( options.base_path.empty() || options.log_level == kC4LogNone ) {
+    if ( options.log_level == kC4LogNone ) {
         // Disabling file logging:
         endFileLogging();
         sDefaultLogFilesLevel = kC4LogNone;
+
+        if ( !options.base_path.empty() ) {
+            auto obs = make_retained<LogFiles>(convertFileOptions(options));
+            stagedDefaultLogFiles.reset(toExternal(std::move(obs)));
+        }
+
+        return true;
+    }  // Post-conditions: options.log_level != kC4LogNone
+
+    if ( options.base_path.empty() ) {
+        // treat it as setLevel
+        c4log_setBinaryFileLevel(options.log_level);
```

Comment:
Author's notes: I handle it this way to achieve some symmetry between path and level. we stage the path when the level is none and I try here to do with the level when the path is empty. What is still not symmetric is that we don't stage the level.
Please offer your suggestions.

Reply:
I don't know as there is a need to stage at all.  Can't you just have the log files set up?  They won't do anything if the level is None right?